### DISCUSSION TITLE
How to avoid to propagate mouse click in an ImGui Window?

### DISCUSSION THREAD

**Discussion by coderc264** [pelicanmapping/osgearth#2372] [Support Q&A] [answered]
Hi all, how can I prevent mouse clicks from being intercepted by an "ObjectIDPicker" when I'm using an "imgui" window?
Now every time I click the mouse on an imgui component the same event is propagated down.

**Comment by jasonbeverage** [comment]
Are you using osgearth_pick or your own application?  If you are using your own application make sure that the you are adding the GUI as the first GUIEventHandler so it can intercept all mouse events like this:
viewer.getEventHandlers().push_front(gui);

I tried with osgearth_imgui and osgearth_pick and click and double click events don't seem to be propagated into the viewer window in those examples.

**Comment by coderc264** [comment, marked as answer]
I solved it, using the ImGui::GetIO().WantCaptureMouse in the ObjectIDPicker function call back
